pkg/antctl/commandbundle.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package antctl
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/server/mux"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/monitor"
+)
+
+// CommandBundle organizes CommandOptions and applies them to a antctl client or server.
+type CommandBundle struct {
+	CommandOptions []CommandOption
+	GroupVersion   *schema.GroupVersion
+	Codec          serializer.CodecFactory
+}
+
+// APIPrefix returns the API prefix of the CLI service according to the GroupVersion.
+func (b *CommandBundle) APIPrefix() string {
+	return path.Join("/apis", b.GroupVersion.Group, b.GroupVersion.Version)
+}
+
+// ApplyToRouter applies the CommandBundle to a router.
+func (b *CommandBundle) ApplyToRouter(r *mux.PathRecorderMux, aq monitor.AgentQuerier, cq monitor.ControllerQuerier) {
```

Comment:
Sorry but is "router" a generic term from somewhere? If not maybe add some comments to explain what is a router.

Reply:
My bad, I think `mux` would be better.